### ISSUE TITLE
Method __str__ of class SummonerSpell doesn't work as intended

### ISSUE THREAD

**Issue by JasonMendoza2008** [meraki-analytics/cassiopeia#433]
Minimal reproducible example:

```py
def print_summ_spell(name: str, region: str) -> None:
    summoner = Summoner(name=name, region=region)
    match_history = summoner.match_history
    match = match_history[0]
    p: cassiopeia.core.match.Participant = match.participants[summoner]
    print(p.summoner_spell_f)


if __name__ == "__main__":
    print_summ_spell(name="veryveryfast", region="EUW")
```

returns

`SummonerSpell(name='?', id=4, region='EUW')`

because this if statement was not entered
```py
        if hasattr(self._data[SummonerSpellData], "name"):
            name = self.name
```
Why not just do `if hasattr(self, "name"): name=self.name`?

**Comment by jjmaldonis** [comment]
This is working as intended because the summoner spell data hasn't been loaded yet.

**Comment by JasonMendoza2008** [comment]
how do I load the summoner spell data? The attribute .name does return what I want but not the print? 
Sorry i'm new to this!

**Comment by JasonMendoza2008** [comment]
@jjmaldonis 
how do I load the summoner spell data? The attribute .name does return what I want but not the print? 
Sorry i'm new to this!

**Comment by jjmaldonis** [comment]
No worries, the following code should print it out:

```
def print_summ_spell(name: str, region: str) -> None:
    summoner = Summoner(name=name, region=region)
    match_history = summoner.match_history
    match = match_history[0]
    p: cassiopeia.core.match.Participant = match.participants[summoner]
    name = p.summoner_spell_f.name  # Accessing the `.name` attribute will load the summoner spell data, and then it will be displayed in the following print
    print(p.summoner_spell_f)


if __name__ == "__main__":
    print_summ_spell(name="veryveryfast", region="EUW")
```

**Comment by JasonMendoza2008** [comment]
Thanks @jjmaldonis! Is this what is called `Ghost Loading` in the documentation?

**Comment by jjmaldonis** [comment]
Yes that's exactly correct. The objects act as "ghosts" before their data is loaded, and they have an incorporeal feeling to them when using them. You noticed this incorporeal-ness in your code above. The object exists, but for some reason it didn't have all the information you expected it to have. This is a ghost.

It's useful because you can still use the objects in the same way, regardless of whether or not they are ghosts or fully-loaded objects. As an example, let's say you have a database that stores summoner names and their puuids. The following code _will not_ make a request to the Riot API to get the summoner info, but you can still interact with the `Summoner` object exactly the same way as if the data did come from the Riot API.

```
my_name, my_puuid = load_my_summoner_from_database()
me = Summoner(name=name, puuid=puuid)

match_history = me.match_history
for match in match_history:
    for p in match.participants:
        if p.summoner.puuid == me.puuid:
            print(f"I played {p.champion.name}")

# And the important part is that Cass never makes a call to the Riot API to get the summoner's info.
# This code will only make calls to the match history, match, and ddragon-champions endpoints.
```

**Comment by JasonMendoza2008** [comment]
Thanks!

Although I'm not sure how there will be no requests to the Riot API if I'm going through the match history?
In your example, `load_my_summoner_from_database()` only stores a string and an int right?

**Comment by JasonMendoza2008** [comment]
Thanks @jjmaldonis!

Although I'm not sure how there will be no requests to the Riot API if I'm going through the match history?
In your example, `load_my_summoner_from_database()` only stores a `string` and an `int` right? If no requests are made, where is the info fetched?

**Comment by jjmaldonis** [comment]
So the code above will make requests to the Riot API for the match history, match, and ddragon-champions endpoints, but it _will not_ make a request to the summoner endpoint despite writing `me = Summoner(name=name, puuid=puuid)`. This line of code does not trigger a call to any of the summoner endpoints.

The summoner endpoint will only be called if something besides for the summoner's `name` or `puuid` is used, which is never the case in this code.

**Comment by JasonMendoza2008** [comment]
Ohhh ok thanks!!!
